crates/pallet-domains/src/block_tree.rs

```diff
@@ -215,7 +247,8 @@ pub(crate) fn verify_execution_receipt<T: Config>(
             );
             Err(Error::PrunedReceipt)
         }
-        ReceiptType::Accepted(_) | ReceiptType::Stale => Ok(()),
+        ReceiptType::Rejected(rejected_receipt_type) => Err(rejected_receipt_type.into()),
```

Comment:
please the log the error here like we do above and I think we can combine all the Rejected match branches into one branch

Reply:
The error log is necessary for the above branch because we are unexpected to reach the `InFutureReceipt` or `PrunedReceipt` here, these receipts should be rejected earlier by the previous check as the log declare. For other rejected ER, they are expected in some situation and already logged in the `Bad bundle ...`